src/components/DappBrowser/handleProviderRequest.ts

```diff
@@ -344,9 +343,9 @@ export const handleProviderRequestApp = ({ messenger, data, meta }: { messenger:
     checkRateLimit,
     onSwitchEthereumChainNotSupported,
     onSwitchEthereumChainSupported,
-    getProvider: chainId => getProvider({ chainId: chainId as number }) as unknown as Provider,
+    getProvider: ({ chainId }) => getProvider({ chainId: chainId as number }),
```

Comment:
this is the main fix:  as part of the network > chainId changes, one of the functions got switched from passing in an object param to passing in the chainId directly.